### PR TITLE
Update Keycloak Test Module to support code flow tests

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR does a few minor improvements to the newly introduced Keycloak test module nicely done by @cemnura :

- Customizes realm, client ids, makes sure it can support both `service` and `web-app` tests, work over plain HTTP
- Updated the docs
- Updated the dev code flow tests in `extensions/oidc/deployment` to use it

### CODE REVIEW

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc [hunk 1]
```diff
@@ -576,15 +576,10 @@ Please see link:security-openid-connect-client#token-propagation[Token Propagati
 [[integration-testing]]
 == Testing
 
-Add the following dependencies to your test project:
+Start with adding the following dependencies to your test project:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:579]:
```suggestion
Start by adding the following dependencies to your test project:
```

File: docs/src/main/asciidoc/security-openid-connect.adoc [hunk 2]
```diff
@@ -434,26 +434,39 @@ Please see link:security-openid-connect-client#token-propagation[Token Propagati
 [[integration-testing]]
 == Testing
 
-=== Wiremock
-
-Add the following dependencies to your test project:
+Start with adding the following dependencies to your test project:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/security-openid-connect.adoc:437]:
```suggestion
Start by adding the following dependencies to your test project:
```

**Review by gastaldi — APPROVED** [review]
LGTM, just some minor changes

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:579]: @gastaldi Thanks, funny, my first try was `Start from` :-)

**Comment by geoand** [comment]
@sberyozkin Hi,

can you please squash the commits?

Thanks

**Comment by sberyozkin** [comment]
@geoand Hi Georgious, oh sorry, will do, by the way, I wanted to give it a quick try and see if this test module can also be used to test `quarkus-oidc-client`, so I'll look into it a bit and make sure the commits are squashed as well, thanks

**Comment by geoand** [comment]
Cool, thanks for the info

**Review by pedroigor — APPROVED** [review]

**Comment by sberyozkin** [comment]
@gastaldi @pedroigor Just pushed another update - where I switched `oidc-client-filter/deployment` to use the new test module and also did a minor doc update.
@gsmet thanks very much for your help with github related issues :-) (for the record, I've messed up the history by accidentally rebasing while pulling George's updates) - please cancel it if it is too much for CI at this point - I'll restart later - not an urgent PR

**Comment by sberyozkin** [comment]
OK, it is all green so I'll go ahead with the merge :-)

**Pull request merged**